src/pyspartn/spartnmessage.py

```diff
@@ -374,7 +374,11 @@ def _set_attribute_single(
             attlen = self._getvarlen(key, index)
         if not self._scaling:
             res = 0
-        val = bitsval(self._payload, offset, attlen)
+        try:  # TODO temporary DEBUG of payload failure
+            val = bitsval(self._payload, offset, attlen)
+        except SPARTNMessageError as err:
+            print(self)
```

Comment:
remember to remove this print in the final version

Reply:
noted